sui_core/src/consensus_adapter.rs

```diff
@@ -115,6 +115,8 @@ pub struct ConsensusSubmitter {
     committee: Committee,
     /// A channel to notify the consensus listener of new transactions.
     tx_consensus_listener: Sender<ConsensusInput>,
+    /// The maximum duration to wait from consensus before aborting the transaction.
+    max_delay: Duration,
```

Comment:
here: https://github.com/MystenLabs/sui/blob/d2a0910339867e7c6a2e304e51ee2d53b77ff583/sui_core/src/consensus_adapter.rs#L179

Reply:
No you can't abort, the client needs to retry until its transaction gets sequences. It can try another validator, submitting to f+1 validator, proposing more gas

Reply:
For most people, the Gateway service will take care of this kind of retrials